File: reverse_string/reverse_string.rb
class Solution
  def reverse_string(s)
    s.reverse!
  end
end

require 'rspec'

RSpec.describe Solution do
  it 'reverses the string' do
    solution = Solution.new
    arr1 = ['h', 'e', 'l', 'l', 'o']
    arr2 = ['H', 'a', 'n', 'n', 'a', 'h']

    solution.reverse_string(arr1)
    solution.reverse_string(arr2)

    expect(arr1).to eq(['o', 'l', 'l', 'e', 'h'])
    expect(arr2).to eq(['h', 'a', 'n', 'n', 'a', 'H'])
  end
end

File: reverse_string/reverse_string.exs
defmodule Solution do
  def reverse_string(s), do: Enum.reverse(s)
end

defmodule SolutionTest do
  use ExUnit.Case
  doctest Solution

  test "reverses the string" do
    assert Solution.reverse_string(['h', 'e', 'l', 'l', 'o']) == ['o', 'l', 'l', 'e', 'h']
    assert Solution.reverse_string(['H', 'a', 'n', 'n', 'a', 'h']) == ['h', 'a', 'n', 'n', 'a', 'H']
  end
end
